### PR TITLE
Remove all references to TS_USE_TLS_SNI.

### LINKED ISSUE #2431: Remove TS_USE_TLS_SNI
> Removed the check on TS_USE_TLS_SNI because 7.X and on don't support versions of openSSL that don't have SNI.

#2427

### PR DESCRIPTION
**Pull request by calavera**
ATS 7.X and on don't support versions of openSSL that don't have SNI.

Fixes #2431

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci]

**Comment by scw00** [comment]
[approve ci]

**Comment by zwoop** [comment]
i'm ok with this, but you missed one place (at least): build/crypto.m4

**Comment by zwoop** [comment]
@calavera ping ? :-)

**Comment by zwoop** [comment]
[approve ci].

File: build/crypto.m4 [hunk 1]
```diff
@@ -86,7 +86,6 @@ AC_DEFUN([TS_CHECK_CRYPTO_ALPN], [
 
 AC_DEFUN([TS_CHECK_CRYPTO_SNI], [
```
Comment (zwoop) [inline build/crypto.m4]: Do we need this function at all? Can't we just nuke the whole thing?

**Comment by calavera** [comment]
@zwoop I've removed the function all together, I think there are no other references.

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
Yes, we should definitely remove that define.

**Comment by masaori335** [comment]
[approve ci]

**Comment by masaori335** [comment]
[approve ci]

**Comment by shinrich** [comment]
[approve ci]

**Comment by masaori335** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

**Comment by shinrich** [comment]
[approve ci fedora]

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
[approve ci CentOS]
[approve ci Ubuntu]
[approve ci Fedora]
[approve ci autest]

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by shinrich** [comment]
[approve fedora]

**Comment by shinrich** [comment]
[approve autest]

**Comment by shinrich** [comment]
[approve ci fedora]

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by shinrich** [comment]
@zwoop did the autest build change to openssl 1.1?  The compilation error looks like we aren't detecting openssl 1.1 correctly but compiling against openssl 1.1.

@calavera perhaps you need to rebase against master?

**Comment by SolidWallOfCode** [comment]
[approve ci fedora]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
[approve ci fedora]

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by calavera** [comment]
I rebased with master 🤞

**Comment by calavera** [comment]
Rebased again to fix a merge conflict. @zwoop, @shinrich do you mind kicking CI?

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by shinrich** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

File: build/crypto.m4 [hunk 2]
```diff
@@ -119,14 +119,13 @@ AC_DEFUN([TS_CHECK_CRYPTO_SNI], [
     enable_tls_sni=no
   ])
 
-  AC_CHECK_FUNCS(SSL_get_servername, [], [enable_tls_sni=no])
+  AC_CHECK_FUNCS(SSL_get_servername, [], [enable_tls_sni])
```
Comment (bryancall) [inline build/crypto.m4]: Shouldn't we remove this also?  There is really no reason to have enable_tls_sni now.

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
Opened #3136 to fix the build problems on this PR and added a commit to completely remove the SNI checks.  calavera is set as the author on the first commit on #3136

**Pull request closed without merging**